feat(java): Migration from 6.x.x to 7.0.0 (#8601)
getsentry/sentry-docs: src/platforms/java/common/performance/instrumentation/okhttp.mdx

</dependency>
```

```groovy {tabTitle:Gradle}
implementation 'io.sentry:sentry-okhttp:{{@inject packages.version('sentry.java.okhttp', '7.0.0') }}'
```

```scala {tabTitle: SBT}
libraryDependencies += "io.sentry" % "sentry-okhttp" % "{{@inject packages.version('sentry.java.okhttp', '7.0.0') }}"
```

For other dependency managers, see the [central Maven repository](https://search.maven.org/artifact/io.sentry/sentry-okhttp).

## Configure

Configuration should happen once you create your `OkHttpClient` instance.

```kotlin
import okhttp3.OkHttpClient
import io.sentry.okhttp.SentryOkHttpEventListener
import io.sentry.okhttp.SentryOkHttpInterceptor

private val client = OkHttpClient.Builder()
  .addInterceptor(SentryOkHttpInterceptor())
  .eventListener(SentryOkHttpEventListener())
  .build()
```

```java
import okhttp3.OkHttpClient;
import io.sentry.okhttp.SentryOkHttpEventListener;
import io.sentry.okhttp.SentryOkHttpInterceptor;

private final OkHttpClient client = new OkHttpClient.Builder()
  .addInterceptor(new SentryOkHttpInterceptor())
  .eventListener(new SentryOkHttpEventListener())
  .build();
```

> The SDK uses the `SentryOkHttpInterceptor` to support [Distributed Tracing](/product/sentry-basics/tracing/distributed-tracing/) and creates a span for any http call.

Reviewer: ```suggestion
The SDK uses the `SentryOkHttpInterceptor` to support [Distributed Tracing](/product/sentry-basics/tracing/distributed-tracing/) and creates a span for any HTTP call.
```

> The SDK can give more in-depth information through the `SentryOkHttpEventListener` by creating a span for each operation reported by OkHttp, including: dns setup, proxy selection, http connect, ssl setup, send request headers, send request body, receive response headers, and receive response body.

Reviewer: ```suggestion
The SDK can give more in-depth information through the `SentryOkHttpEventListener` by creating a span for each operation reported by OkHttp, including DNS setup, proxy selection, HTTP connect, SSL setup, send request headers, send request body, receive response headers, and receive response body.
```

import okhttp3.Request;
import okhttp3.Response;

class CustomBeforeSpanCallback implements SentryOkHttpInterceptor.BeforeSpanCallback {
  public ISpan execute(ISpan span, Request request, Response response) {
    if (request.getUrl().toUri().toString().contains("/admin")) {
      return null;
    } else {
      return span;
    }
  }
}
```

The callback instance must be set on the `SentryOkHttpInterceptor` once you create your `OkHttpClient` instance.

```kotlin
import okhttp3.OkHttpClient
import io.sentry.okhttp.SentryOkHttpEventListener
import io.sentry.okhttp.SentryOkHttpInterceptor

private val client = OkHttpClient.Builder()
  .eventListener(SentryOkHttpEventListener())
  .addInterceptor(SentryOkHttpInterceptor(CustomBeforeSpanCallback()))
  .build()
```

```java
import okhttp3.OkHttpClient;
import io.sentry.okhttp.SentryOkHttpEventListener;
import io.sentry.okhttp.SentryOkHttpInterceptor;

private final OkHttpClient client = new OkHttpClient.Builder()
  .eventListener(new SentryOkHttpEventListener())
  .addInterceptor(new SentryOkHttpInterceptor(new CustomBeforeSpanCallback()))
  .build();
```

## Using Multiple Event Listeners

> The event listener can propagate calls to another `EventListener` or `EventListener.Factory` that's set on the `SentryOkHttpEventListener` once you create your `OkHttpClient` instance.

Reviewer: Is this edit correct? Since this is the first time we reference an event listener on the page, I think it's better to specify.

```suggestion
The integration's event listener can propagate calls to another `EventListener` or `EventListener.Factory` that's set on the `SentryOkHttpEventListener` once you create your `OkHttpClient` instance.
```

    failedRequestStatusCodes = listOf(HttpStatusCodeRange(400, 599))))
  .build()
```

HTTP client errors from every target (`.*` regular expression) are automatically captured, but you can change this behavior by setting the `failedRequestTargets` option with either a regular expression or a plain `String`. A plain string must contain at least one of the items from the list. Plain strings don't have to be full matches, meaning the URL of a request is matched when it contains a string provided through the option.

```kotlin
import okhttp3.OkHttpClient
import io.sentry.okhttp.SentryOkHttpEventListener
import io.sentry.okhttp.SentryOkHttpInterceptor
import io.sentry.HttpStatusCodeRange

private val client = OkHttpClient.Builder()
  .eventListener(SentryOkHttpEventListener())
  .addInterceptor(SentryOkHttpInterceptor(
    captureFailedRequests = true,
    failedRequestTargets = listOf("myapi.com")))
  .build()
```

By default, error events won't contain any PII data, such as `Headers` and `Cookies`, but you can change this behavior by setting the `sendDefaultPii` option to `true`:

```kotlin
Sentry.init { options ->
  options.isSendDefaultPii = true
}
```

> Those events are searchable and you can set alerts on them if you use the `http.url` and `http.status_code` properties. Learn more in our full [Searchable Properties](/product/sentry-basics/search/searchable-properties/) documentation.

Reviewer: Which events are "those"? Took my best guess with this edit :)

```suggestion
HTTP client errors sent to Sentry are searchable and you can set alerts on them if you use the `http.url` and `http.status_code` properties. Learn more in our full [Searchable Properties](/product/sentry-basics/search/searchable-properties/) documentation.
```
